src/python_testing/TC_SC_4_3.py

```diff
@@ -0,0 +1,407 @@
+            try:
+                # Attempt to create an IPv6 address object. If successful, this is an IPv6 address.
+                ipaddress.IPv6Address(address)
+                return True, "At least one IPv6 address is present."
+            except ipaddress.AddressValueError:
+                # If an AddressValueError is raised, the current address is not a valid IPv6 address.
+                return False, f"Invalid IPv6 address encountered: {address}, provided addresses: {addresses}"
+        return False, "No IPv6 addresses found."
+
+    @staticmethod
+    def extract_ipv6_address(text):
+        items = text.split(',')
+        return items[-1]
+
+    @staticmethod
+    def verify_hostname(hostname: str) -> bool:
+        # Remove '.local' or '.local.' suffix if present
+        if hostname.endswith('.local'):
+            hostname = hostname[:-6]
+        elif hostname.endswith('.local.'):
+            hostname = hostname[:-7]
+
+        # Regular expression to match an uppercase hexadecimal string of 12 or 16 characters
+        pattern = re.compile(r'^[0-9A-F]{12}$|^[0-9A-F]{16}$')
+        return bool(pattern.match(hostname))
+
+    @async_test_body
+    async def test_TC_SC_4_3(self):
+        supports_icd = None
+        supports_lit = None
+        active_mode_threshold_ms = None
+        instance_name = None
+
+        # *** STEP 1 ***
+        # DUT is commissioned on the same fabric as TH.
+        self.step(1)
+
+        # *** STEP 2 ***
+        # TH reads from the DUT the ServerList attribute from the Descriptor cluster on EP0. If the ICD Management
+        # cluster ID (70,0x46) is present in the list, set supports_icd to true, otherwise set supports_icd to false.
+        self.step(2)
+        ep0_servers = await self.get_descriptor_server_list()
+
+        # Check if ep0_servers contains the ICD Management cluster ID (0x0046)
+        supports_icd = Clusters.IcdManagement.id in ep0_servers
+        logging.info(f"\n\n\tsupports_icd: {supports_icd}\n\n")
+
+        # *** STEP 3 ***
+        # If supports_icd is true, TH reads ActiveModeThreshold from the ICD Management cluster on EP0 and saves
+        # as active_mode_threshold.
+        self.step(3)
+        if supports_icd:
+            active_mode_threshold_ms = await self.get_idle_mode_threshhold_ms()
+        logging.info(f"\n\n\tactive_mode_threshold_ms: {active_mode_threshold_ms}\n\n")
+
+        # *** STEP 4 ***
+        # If supports_icd is true, TH reads FeatureMap from the ICD Management cluster on EP0. If the LITS feature
+        # is set, set supports_lit to true. Otherwise set supports_lit to false.
+        self.step(4)
+        if supports_icd:
+            feature_map = await self.get_icd_feature_map()
+            LITS = Clusters.IcdManagement.Bitmaps.Feature.kLongIdleTimeSupport
+            supports_lit = bool(feature_map & LITS == LITS)
+            logging.info(f"\n\n\tkLongIdleTimeSupport set: {supports_lit}\n\n")
```

Comment:
These logs are fine, but can you remove some of the newlines here? It might make the logs look weird.

Reply:
Done.